velox/exec/RowContainer.h

```diff
@@ -728,6 +726,48 @@ class RowContainer {
     }
   }
 
+  template <bool useRowNumbers, typename T>
+  static void extractValues(
+      const char* FOLLY_NONNULL const* FOLLY_NONNULL rows,
+      folly::Range<const vector_size_t*> rowNumbers,
+      int32_t numRows,
+      int32_t offset,
+      int32_t resultOffset,
+      FlatVector<T>* FOLLY_NONNULL result,
+      bool withNulls = false,
```

Comment:
It feels a bit odd to have such an important top-level parameter appear so late in the arguments. I agree with the removal of code-duplication, the original function signatures were better imo.

Do you have any other ideas given that line of thinking ?

Reply:
Thanks for pointing out the importance of the parameter placement. I agree that the original function signatures are better.
Given this, how about templatizing `withNulls`?
```
template <bool withNulls, bool useRowNumbers, typename T>
static void extractValues(
    const char* FOLLY_NONNULL const* FOLLY_NONNULL rows,
    folly::Range<const vector_size_t*> rowNumbers,
    int32_t numRows,
    int32_t offset,
    int32_t resultOffset,
    FlatVector<T>* FOLLY_NONNULL result,
    int32_t nullByte = 0,
    uint8_t nullMask = 0);
```
(Please see the complete function definition in my reply to masha).
This makes `withNulls` the top-most parameter, `nullByte`/`nullMask` at the tail is still no preferable, but that's because they have default value.